client/fingerprint/env_aws_test.go

```diff
@@ -268,48 +142,99 @@ func TestNetworkFingerprint_AWS_network(t *testing.T) {
 		request := &FingerprintRequest{Config: cfg, Node: node}
 		var response FingerprintResponse
 		err := f.Fingerprint(request, &response)
-		if err != nil {
-			t.Fatalf("err: %v", err)
-		}
+		require.NoError(t, err)
 
 		assertNodeAttributeContains(t, response.Attributes, "unique.network.ip-address")
 
-		if response.NodeResources == nil || len(response.NodeResources.Networks) == 0 {
-			t.Fatal("Expected to find Network Resources")
-		}
+		require.NotNil(t, response.NodeResources)
+		require.Len(t, response.NodeResources.Networks, 1)
 
 		// Test at least the first Network Resource
 		net := response.NodeResources.Networks[0]
-		if net.IP == "" {
-			t.Fatal("Expected Network Resource to have an IP")
-		}
-		if net.CIDR == "" {
-			t.Fatal("Expected Network Resource to have a CIDR")
-		}
-		if net.Device == "" {
-			t.Fatal("Expected Network Resource to have a Device Name")
-		}
-		if net.MBits != 10 {
-			t.Fatalf("Expected Network Resource to have speed %d; got %d", 10, net.MBits)
-		}
+		require.NotEmpty(t, net.IP, "Expected Network Resource to have an IP")
+		require.NotEmpty(t, net.CIDR, "Expected Network Resource to have a CIDR")
+		require.NotEmpty(t, net.Device, "Expected Network Resource to have a Device Name")
+		require.Equal(t, 10, net.MBits)
 	}
 }
 
-func TestNetworkFingerprint_notAWS(t *testing.T) {
-	os.Setenv("AWS_ENV_URL", "http://127.0.0.1/latest/meta-data/")
-	f := NewEnvAWSFingerprint(testlog.HCLogger(t))
-	node := &structs.Node{
-		Attributes: make(map[string]string),
-	}
+/// Utility functions for tests
 
-	request := &FingerprintRequest{Config: &config.Config{}, Node: node}
-	var response FingerprintResponse
-	err := f.Fingerprint(request, &response)
-	if err != nil {
-		t.Fatalf("err: %v", err)
+func startFakeEC2Metadata(t *testing.T) (endpoint string, cleanup func()) {
+	routes := routes{}
+	if err := json.Unmarshal([]byte(aws_routes), &routes); err != nil {
+		t.Fatalf("Failed to unmarshal JSON in AWS ENV test: %s", err)
 	}
 
-	if len(response.Attributes) > 0 {
-		t.Fatalf("Should not apply")
-	}
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		for _, e := range routes.Endpoints {
+			if r.RequestURI == e.Uri {
+				w.Header().Set("Content-Type", e.ContentType)
+				fmt.Fprintln(w, e.Body)
+			}
+		}
+	}))
+
+	return ts.URL + "/latest", ts.Close
+}
+
+type routes struct {
+	Endpoints []*endpoint `json:"endpoints"`
+}
+type endpoint struct {
+	Uri         string `json:"uri"`
+	ContentType string `json:"content-type"`
+	Body        string `json:"body"`
 }
+
+const aws_routes = `
```

Comment:
Not obvious to me why we use json string here instead of writing the struct values directly?